### PR TITLE
Atlas connector

### LINKED ISSUE #2908: Ingestion - Apache Atlas Connector
Creating an Ingestion Connector for Apache Atlas to fetch metadata for multiple entities.

### PR DESCRIPTION
**Pull request by codingwithabhi**
### Describe your changes :

I worked on adding Atlas Services(Kafka, RDBMS, Lineage) into a single file.
fix #2908 
#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.

@ayush-shah

### CODE REVIEW

**Comment by harshach** [comment]
@ayush-shah @codingwithabhi we should provide a config that can take a mapping file from the users
1. Which will users to specify how their json is structured in Atlas
2. Atlas allows user model their own structure for metadata

This file will allow them to give us that mapping and we can use that config to fetch metadata and parse it.

File: ingestion/src/metadata/ingestion/source/atlas.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        bootstrap_servers = "http://localhost:9092"
+        self.message_service = get_messaging_service_or_create(
+            config.service_name,
+            MessagingServiceType.Kafka.name,
+            schema_registry_url,
+            bootstrap_servers.split(","),
+            metadata_config,
+        )
+        self.atlas_client = AtlasClient(config)
+        f = open(self.config.entity_types)
```
Comment (ayush-shah) [inline ingestion/src/metadata/ingestion/source/atlas.py:82]: check if the file exists at the given path

File: ingestion/src/metadata/ingestion/source/atlas.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+    def _parse_table_entity(self):
+        for table in self.tables:
+            table_entity = self.atlas_client.get_entity(table)
+            tbl_entities = table_entity["entities"]
+            for tbl_entity in tbl_entities:
+                try:
+                    tbl_columns = self._parse_table_columns(table_entity, tbl_entity)
+                    tbl_attrs = tbl_entity["attributes"]
+                    db_entity = tbl_entity["relationshipAttributes"]["db"]
```
Comment (ayush-shah) [inline ingestion/src/metadata/ingestion/source/atlas.py:147]: Add the value from the mapping file here

Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/source/atlas.py:147]: for Database

File: ingestion/src/metadata/ingestion/source/atlas.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+                    om_table_entity = CreateTableRequest(
+                        name=table_name,
+                        columns=tbl_columns,
+                        database=db.id,
+                        fullyQualifiedName=fqn,
+                    )
+
+                    created_table = self.metadata.create_or_update(om_table_entity)
+                    yield created_table
```
Comment (ayush-shah) [inline ingestion/src/metadata/ingestion/source/atlas.py:174]: we don't need this on the source side

**Comment by harshach** [comment]
@codingwithabhi rebase your PR and merge it in

File: ingestion/src/metadata/ingestion/source/atlas.py [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+    def prepare(self):
+        for key in self.config.entity_types["Table"].keys():
+            self.tables[key] = self.atlas_client.list_entities(entity_type=key)
+        for key in self.config.entity_types["Topic"]:
+            self.topics[key] = self.atlas_client.list_entities(entity_type=key)
+
+    def next_record(self):
+        if self.tables:
+            for _, entity in enumerate(self.tables):
```
Comment (ayush-shah) [inline ingestion/src/metadata/ingestion/source/atlas.py:101]: if you don't need index, you can remove enumerate

File: ingestion/src/metadata/ingestion/source/atlas.py [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+        for table in entity:
+            table_entity = self.atlas_client.get_entity(table)
+            tbl_entities = table_entity["entities"]
+            for tbl_entity in tbl_entities:
+                try:
+                    tbl_columns = self._parse_table_columns(
+                        table_entity, tbl_entity, name
+                    )
+                    tbl_attrs = tbl_entity["attributes"]
+                    db_entity = tbl_entity["relationshipAttributes"]["db"]
```
Comment (ayush-shah) [inline ingestion/src/metadata/ingestion/source/atlas.py:145]: Please handle relationshipAttributes and mapping from json file instead of `["db"]`

**Review by ayush-shah — APPROVED** [review]

**Pull request merged**